articles/integrating-b2c-mpesa-api-using-node-js/index.md

```diff
@@ -0,0 +1,299 @@
+
+B2C Mpesa API enables software developers to integrate the functionality of merchants sending payments to their customers in their applications.
+
+### Goals
+
+In this article, we will integrate the B2C Mpesa API to a Node.js RESTful API. We will consume the API using [swagger](https://swagger.io/).
+
+### Prerequisites
+
+To follow along in this article, it is helpful to have the following:
+
+- [Node.js](https://nodejs.org/en/) installed on your computer.
+- Some basic knowledge working with JavaScript.
+- A text editor installed.
+
+### Overview
+
+- [Setting up the development server](#setting-up-the-development-server)
+- [Getting an access token](#getting-an-access-token)
+- [B2c API](#b2c-api)
+
+### Setting up the development server
+
+This article assumes that you have a Safaricom developer account.
+
+If you don't, go through these [steps](/engineering-education/lipa-na-mpesa-online/#creating-a-safaricom-developer-account).
+
+Also, ensure that you have created at least one application from your developer portal. If you don't have any, follow this [guidelines](/engineering-education/lipa-na-mpesa-online/#creating-an-app).
+
+To get started, we will clone the project skeleton from [here](https://github.com/Roba-W/nodejs-b2c-mpesa-api-starterkit). With all the basic configuration done on the skeleton, our focus throughout the article will be implementing the core functionalities.
+
+To start with, we need to install the following dependencies:
+
+- [axios](https://www.npmjs.com/package/axios): For handling the requests to the Mpesa API.
+- [dotenv](https://www.npmjs.com/package/dotenv): For loading the environmental variables.
+- [express](https://www.npmjs.com/package/express): For providing a faster, and easier to work with set up for the restful API.
+- [ngrok](https://www.npmjs.com/package/ngrok): For exposing our localhost server.
+- [swagger-ui-express](https://www.npmjs.com/package/swagger-ui-express): For serving the API docs to the user interface.
+
+To do that, open your cloned project skeleton in your text editor. Open the terminal of your text editor and run the following command:
+
+```bash
+npm install
+```
+
+Having installed the dependencies, you are okay to move on to the next step.
+
+We will be working from the `src/controllers/Mpesa.js` file.
+
+### Getting an access token
+
+For any request to the Mpesa API, an access token is a requirement. It forms the basis for authentication.
+
+To implement the functionality, we will edit the `getAccessToken()` method as follows:
+
+```js
+//get access token.
+async getAccessToken(req,res,next){
+    // get data from headers by destructuring
+    let {consumerkey,consumersecret} = req.headers;
+    // request url
+    let url = "https://sandbox.safaricom.co.ke/oauth/v1/generate?grant_type=client_credentials";
+
+    // base64 encoded string from a buffer
+    let buf = new Buffer.from(`${consumerkey}:${consumersecret}`).toString("base64");
```

Comment:
what does the variable buf store? Could you inform the reader

Reply:
done